File: api/app/schemas/analytics.py
from pydantic import BaseModel


class DailyMessageCount(BaseModel):
    date: str
    count: int


class DailyTokenUsage(BaseModel):
    date: str
    prompt_tokens: int
    completion_tokens: int


class ToolUsageCount(BaseModel):
    tool_name: str
    count: int


class AnalyticsTotals(BaseModel):
    conversations: int
    messages: int
    prompt_tokens: int
    completion_tokens: int


class AnalyticsSummary(BaseModel):
    messages_per_day: list[DailyMessageCount]
    tokens_per_day: list[DailyTokenUsage]
    tool_usage: list[ToolUsageCount]
    totals: AnalyticsTotals

File: web/src/api/analytics.ts
import { API_PREFIX, apiFetch } from '@/lib/api'

export interface DailyMessageCount {
  date: string
  count: number
}

export interface DailyTokenUsage {
  date: string
  prompt_tokens: number
  completion_tokens: number
}

export interface ToolUsageCount {
  tool_name: string
  count: number
}

export interface AnalyticsTotals {
  conversations: number
  messages: number
  prompt_tokens: number
  completion_tokens: number
}

export interface AnalyticsSummary {
  messages_per_day: DailyMessageCount[]
  tokens_per_day: DailyTokenUsage[]
  tool_usage: ToolUsageCount[]
  totals: AnalyticsTotals
}

export function getAnalyticsSummary(days = 14) {
  return apiFetch<AnalyticsSummary>(`${API_PREFIX}/analytics/summary?days=${days}`)
}
